### PR TITLE
Don't obtain the length of the pexecs from a potentially crashed [] pexec

### LINKED ISSUE #352: In the event that the first process execution crashed, the plot script cannot run
```
    first_key = data['data'].keys()[0]
    first_mc = data['data'][data['data'].keys()[0]].keys()[0]
    first_pexec = data['data'][first_key][first_mc][0]
    if options. window_size >= len(first_pexec):  # Assume all execs are the same length
        fatal_error('Window size (%d) must be strictly less than the number '
                    'of iterations in each process execution (%d). Use '
                    '-w or --window on the command line.' %
                    (options.window_size, len(first_pexec)))
```

Is broken since a crash would put in the results file, an empty list of length 0. This means there is no `-w` that you could use to make the script run.

### PR DESCRIPTION
**Pull request by vext01**
Fixes #352, but blocked on #351 for proper testing. DONT MERGE YET.

### CODE REVIEW

**Comment by vext01** [comment]
#351 is merged, so ready for review.

Tested on small examples. Should probably test on some real data. Will do that now.

File: bin/plot_krun_results [hunk 1]
```diff
@@ -1357,14 +1357,29 @@ if __name__ == '__main__':
...
+            for bench_data in key_data.itervalues():
+                for pexec in bench_data:
+                    if pexec == []:
+                        # indicates a crash
+                        continue
+                    else:
+                        iter_lens = len(pexec)
+                        raise StopIteration()  # to break out of all loops at once
+        else:
+            print("couldn't find a non-crashing pexec")
```
Comment (snim2) [inline bin/plot_krun_results]: We've used single quotes in the rest of the script.

**Comment by snim2** [comment]
OK, this looks sensible, I'll give it a test now.

File: bin/plot_krun_results (hunk 1 above)
Comment (vext01) [reply bin/plot_krun_results]: Ah, I can never get used to that. fixed.

**Comment by snim2** [comment]
OK, this seems good to me, squash away!

**Comment by vext01** [comment]
splat.

**Comment by vext01** [comment]
I plotted the data from bencher6 fine using this patch too.

**Comment by snim2** [comment]
OK, cool. I was using b3 v0.7 and your crashed test case.

**Review by snim2 — APPROVED** [review]

**Pull request merged**